buildSrc/src/main/java/VersioningPlugin.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2014-2021 The Android Password Store Authors. All Rights Reserved.
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+
+
+import com.android.build.gradle.internal.plugins.AppPlugin
+import com.github.zafarkhaja.semver.Version
+import java.io.OutputStream
+import java.util.Properties
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+private const val VERSIONING_PROP_FILE = "version.properties"
+private const val VERSIONING_PROP_VERSION_NAME = "versioning-plugin.versionName"
+private const val VERSIONING_PROP_VERSION_CODE = "versioning-plugin.versionCode"
+private const val VERSIONING_PROP_COMMENT = """
+This file was automatically generated by 'versioning-plugin'. DO NOT EDIT MANUALLY.
+"""
+
+/**
+ * A Gradle [Plugin] that takes a [Project] with the [AppPlugin] applied and dynamically sets the
+ * versionCode and versionName properties based on values read from a [VERSIONING_PROP_FILE] file in
+ * the [Project.getBuildDir] directory. It also adds Gradle tasks to bump the major, minor, and patch
+ * versions along with one to prepare the next snapshot.
+ */
+@Suppress(
+    "UnstableApiUsage",
+    "NAME_SHADOWING"
+)
+class VersioningPlugin : Plugin<Project> {
+
+    /**
+     * Generate the Android 'versionCode' property
+     */
+    private fun Version.androidCode(): Int {
+        return majorVersion * 1_00_00 +
+            minorVersion * 1_00 +
+            patchVersion
+    }
+
+    /**
+     * Write an Android-specific variant of [this] to [stream]
+     */
+    private fun Version.writeForAndroid(stream: OutputStream) {
+        val newVersionCode = androidCode()
+        val props = Properties()
+        props.setProperty(VERSIONING_PROP_VERSION_CODE, "$newVersionCode")
+        props.setProperty(VERSIONING_PROP_VERSION_NAME, toString())
+        props.store(stream, VERSIONING_PROP_COMMENT)
```

Comment:
Updating this and the `versionProps` field below to use something that's not `java.util.Properties` is desirable due to the date header issue mentioned above.

Reply:
Maybe this is viable as it doesn't require much additional code: https://stackoverflow.com/a/60162749

Reply:
I have been to the page and like a perfectly meme-replicate developer, never scrolled past the accepted answer. Giving this a shot.

Reply:
Yeah it didn't work and I don't care anymore. Let's ship.